Dfe.Data.SearchPrototype/Web/Models/ViewModels/Shared/ScrollablePager.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Dfe.Data.SearchPrototype.Web.Models.ViewModels.Shared
+{
+    /// <summary>
+    /// Provides functionality to allow paging to work in a scrollable manner whereby moving forwards or
+    /// backwards across a given page sequence means the page selected is maintained in the middle of the sequence,
+    /// thus giving the impression that the pages are scrolling forwards or backwards depending on the selection made.
+    /// </summary>
+    public sealed class ScrollablePager : IPager
+    {
+        /// <summary>
+        /// Sets the default number of lower and upper pages to 2. This means that
+        /// if the current page falls within this lower or upper page band, certain
+        /// constraints can be triggered (e.g. display/hide next previous buttons).
+        /// </summary>
+        private const int PageSequencePaddingSize = 2;
+
+        /// <summary>
+        /// Used to determine if FirstPageNumber should be displayed ahead of PageSequence
+        /// Checks if the first page of PageSequence is less than pageSequencePaddingSize(2)
+        /// if < 2 example: 1 2 3 4 5 [no FirstPageNumber, PageSequence (5 buttons, starting with < 2 so already covered FirstPage)
+        /// if == 2 example: 1 2 3 4 5 6 [FirstPageNumber and PageSequence(5 buttons, starting with == 2)]
+        /// if > 2 example: 1 ... 5 6 7 8 9 [FirstPageNumber, Ellipsis, PageSequence(5buttons, starting with > 2)].
+        /// </summary>
+        /// <param name="currentPageNumber">
+        /// The current page selected.
+        /// </param>
+        /// <param name="totalNumberOfPages">
+        /// The total number of pages available.
+        /// </param>
+        /// <returns>
+        /// True or false based on whether the condition is met (i.e. the
+        /// current page falls within the lower paging bounds).
+        /// </returns>
+        public bool IsCurrentPageInLowerPagingBoundary(
+            int currentPageNumber, int totalNumberOfPages) =>
+                GetFirstPageInSequence(currentPageNumber, totalNumberOfPages) < PageSequencePaddingSize;
+
```

Comment:
talking to Asia, 'boundary' is logic indicating whether first page / last page needs to be shown
'threshold' refers to logic for whether ellipses is needed